File: 101_150/119_pascals_triangle_II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        ArrayList<Integer> res = new ArrayList<>(rowIndex + 1);
        
        for (int i=0; i<rowIndex+1; i++) {
            // add 1 at the end
            res.add(1);
            
            for (int j=i-1; j>0; j--) {
                // sum numbers from right to left
                res.set(j, res.get(j) + res.get(j-1));
            }
        }
        
        return res;
    }
}

File: 101_150/119_pascals_triangle_II.c
/**
 * Return an array of size *returnSize.
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* getRow(int rowIndex, int* returnSize) {
    *returnSize = rowIndex + 1;
    int *res = malloc(sizeof(int) * (*returnSize));
    
    res[0] = 1;
    // compute the next value iteratively
    for (int i=0; i<rowIndex; i++) {
        int nextVal = 1.0 * res[i] * (rowIndex - i) / (i + 1);
        res[i+1] = nextVal;
    }
    
    return res;
}
